react/client.ts

```diff
@@ -11,26 +8,21 @@ export const getAddress = (
   ).then((res) => res.json())
 
 export const updateSession = (
+  countryCode: string,
   zipCode: string,
   geoCoordinates: number[],
   pickup?: Pickup
-) => {
-  const facetsValue = `zip-code=${zipCode};coordinates=${geoCoordinates.join(
-    ','
-  )}${pickup ? `;pickupPoint=${pickup.pickupPoint.id}` : ''}`
-
-  // __RUNTIME__.segmentToken is not reliable for the facets. It might not be updated. For this reason we must try to get the info from our custom cookie first
-  // Encode to base64 because ";" is not allowed in cookies
-  setCookie(SHIPPING_INFO_COOKIE, btoa(facetsValue))
-
-  return fetch('/api/sessions', {
+  // eslint-disable-next-line max-params
+) =>
+  fetch('/api/sessions', {
     method: 'POST',
-    body: `{"public":{"facets":{"value":"${facetsValue}"}}}`,
+    body: `{"public":{"facets":{"value":"zip-code=${zipCode};country=${countryCode};coordinates=${geoCoordinates.join(
```

Comment:
⛏️ Perhaps you can build the object above and then use the JSON.stringfy() to transform this object to a string.

Reply:
About that I need this to actually be a string and not an object so when using the JSON.stringfy() it will convert to a JSON object and parse the values to strings inside, the main issue here is that  I need the formatting to respect the string with `=` and `;` so with that I can parse on the other side without breaking much logic.

In that sense I agreed that it's not pretty but It works well and avoid some more steps and simplify this deploy so we can deliver this fast.